### PR TITLE
Add missing stream/child_process declarations

### LINKED ISSUE #129: Flow fails to find nodeJS core module, child_process
/* @flow */

    var cp = require('child_process');

produces error:

    foo.js:3:10,33: child_process 
    Required module not found

Using nodeJS v0.10.32 on OS X together with flow v0.1.1

### PR DESCRIPTION
**Pull request by josephsavona**
Adds/updates declarations for lib/node.js `Readable` and `Writable` stream classes and `child_process` type. The added test is based on the nodejs.org documentation examples.

Resolve #129.

### CODE REVIEW

File: lib/node.js [hunk 1]
```diff
@@ -524,16 +524,26 @@ declare module "readline" {
...
+  pipe(dest: any, options?: any): void;
 }
 
 declare class stream$Readable extends stream$Stream {
   setEncoding(): void;
+  isPaused(): boolean;
+  pause(): void;
+  read(size?: number): ?(string | Buffer);
+  resume(): void;
+  unpipe(dest: any): void;
```
Comment (gabelevi) [inline lib/node.js]: Looks like the argument here is optional, right? https://nodejs.org/api/stream.html#stream_readable_unpipe_destination

File: lib/node.js [hunk 2]
```diff
@@ -524,16 +524,26 @@ declare module "readline" {
...
+  unshift(chunk: any): void;
+  wrap(oldReadable: any): stream$Readable;
 }
 
 declare class stream$Writable extends stream$Stream {
   close(): void;
+  end(): void;
```
Comment (gabelevi) [inline lib/node.js]: I think `end()` has 3 optional arguments

https://nodejs.org/api/stream.html#stream_writable_end_chunk_encoding_callback

File: lib/node.js [hunk 3]
```diff
@@ -524,16 +524,26 @@ declare module "readline" {
...
   finish(): void;
+  setDefaultEncoding(): void;
```
Comment (gabelevi) [inline lib/node.js]: This takes one argument

https://nodejs.org/api/stream.html#stream_writable_setdefaultencoding_encoding

File: lib/node.js [hunk 4]
```diff
@@ -524,16 +524,26 @@ declare module "readline" {
...
+  write(): void;
```
Comment (gabelevi) [inline lib/node.js]: This has one required argument and 2 optional ones

https://nodejs.org/api/stream.html#stream_writable_write_chunk_encoding_callback

**Comment by gabelevi** [comment]
Awesome! This looks great! Thanks for fixing this issue! I had a few nits inline, but other than that it looks perfect :)

**Comment by mroch** [comment]
what's the status on this guy?

**Pull request closed without merging**